lib/job.js

```diff
@@ -283,8 +283,21 @@ Job.prototype.getState = function() {
 Job.prototype.remove = function(){
   var queue = this.queue;
   var job = this;
-  return scripts.remove(queue, this.jobId).then(function(){
-    queue.emit('removed', job);
+
+  return job.takeLock().then(function(lock) {
+    if (!lock) {
+      throw new Error('Cannot get lock for job: ' + job.jobId + '. cannot be removed.');
+    }
+
+    return scripts.remove(queue, job.jobId)
+      .then(function() {
+        queue.emit('removed', job);
+      })
+      .finally(function () {
+        return job.releaseLock().then(function() {
+          return job;
+        });
```

Comment:
I do not think returning job here has any effect, since finally does not "overwrite" the original returned value (undefined in this case).